mobile/native/src/api.rs

```diff
@@ -37,25 +37,28 @@ pub struct Balances {
     pub lightning: u64,
 }
 
-pub fn refresh_wallet_info() -> Result<WalletInfo> {
-    ln_dlc::get_wallet_info()
+/// Assembles the wallet info and publishes wallet info update event
+#[tokio::main(flavor = "current_thread")]
+pub async fn refresh_wallet_info() -> Result<()> {
+    ln_dlc::refresh_wallet_info().await?;
+
+    Ok(())
 }
 
-#[derive(Clone, Debug, Default)]
-pub struct Transaction {
-    // TODO(Restioson): newtype?
-    pub address: String,
+#[derive(Clone, Debug)]
+pub struct WalletHistoryItem {
     pub flow: PaymentFlow,
-    // TODO(Restioson): newtype?
     pub amount_sats: u64,
+    pub timestamp: u64,
+    pub status: Status,
     pub wallet_type: WalletType,
 }
 
-#[derive(Clone, Debug, Default)]
+#[derive(Clone, Debug)]
 pub enum WalletType {
-    OnChain,
-    #[default]
-    Lightning,
+    OnChain { txid: String },
+    Lightning { payment_hash: String },
```

Comment:
I couldn't figure out if there was a way to identify the sender or the receiver of the payment, so it was finally omitted.